go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -376,9 +376,13 @@ func (qre *QueryExecutor) execDDL(conn *StatefulConnection) (*sqltypes.Result, e
 	if err != nil {
 		return nil, err
 	}
-	err = qre.BeginAgain(qre.ctx, conn)
-	if err != nil {
-		return nil, err
+	// Only perform this operation when the connection has transaction open.
+	// TODO: This actually does not retain the old transaction. We should see how to provide correct behaviour to client.
```

Comment:
You can remove the TODO. The DDL specs say that it's an implicit commit. So, this is correct behavior.

Reply:
The idea here it to not start a new transaction as the current existing transaction is committed. Instead VTGate session should clear out transaction details it is holding.